core/src/main/scala/com/codacy/analysis/core/storage/FileDataStorage.scala

```diff
@@ -0,0 +1,62 @@
+package com.codacy.analysis.core.storage
+
+import better.files.File
+import io.circe._
+import io.circe.syntax._
+import org.log4s.{Logger, getLogger}
+
+import scala.util.Try
+
+trait FileDataStorage[T] {
```

Comment:
Do we need this abstraction?

Reply:
We have 2 different file storage: tools and patterns. This abstraction contains the common part related to read and write from files. So, I would say it is useful 🙂